xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/recently-updated.xml

```diff
@@ -0,0 +1,1881 @@
+    #queryDocuments($documentsResponse)
+    #set ($discard = $queryResponses.add($documentsResponse.results))
+  #end
+  #if ($types.contains('blogpost'))
+    #queryBlogposts($blogpostResponse)
+    #set ($discard = $queryResponses.add($blogpostResponse.results))
+  #end
+  #if ($types.contains('attachment'))
+    #queryAttachments($attachmentsResponse)
+    #set ($discard = $queryResponses.add($attachmentsResponse.results))
+  #end
+  #if ($types.contains('comment'))
+    #queryComments($commentsResponse)
+    #set ($discard = $queryResponses.add($commentsResponse.results))
+  #end
+  ## Resolve used data
+  #mergeSortLimitResults($queryResponses $results)
+  #resolveResultsMetadata($results)
+  ## Return
+  #set ($return = $NULL)
+  #setVariable("$return" $results)
+#end
+
+#fetchResults($results)
+
+{{/velocity}}
+
+{{velocity}}
+$xwiki.ssx.use('xwiki:Confluence.Macros.recently-updated')
+##
+#set ($socialThemePreviousAuthor = $NULL)
+##
+{{html clean="false"}}
+  &lt;section class="recently-updated-macro" $widthStyle&gt;
+    #if ("$wikimacro.parameters.hideHeading" != "true")
+      &lt;h2&gt;Recently Updated&lt;/h2&gt;
+    #end
+
+    &lt;ul class="items theme-${theme}"&gt;
+{{/html}}
+
+{{html clean="false" wiki="true"}}
+      #foreach ($result in $results)
+        #if ($theme == 'sidebar')
+          ## Theme sidebar - start
+          &lt;li&gt;
+            &lt;div class="result-item"&gt;
+              [[image:icon:${ICON_BY_TYPE[$result.type]}||class="document-icon"]]
+              &lt;span class="result-title"&gt;
+                &lt;a href="${result.href}"&gt;${result.title_}&lt;/a&gt;
+              &lt;/span&gt;
+              &lt;span class="result-metadata"&gt;
+                &lt;span class="result-last-author-text"&gt;${AUTHORING_TEXT_BY_TYPE[$result.type]}&lt;/span&gt;
+                #displayAuthorAvatarURL(${result.author})
+                $services.date.displayTimeAgo(${result.date})
+              &lt;/span&gt;
+            &lt;/div&gt;
+          &lt;/li&gt;
+          ## Theme sidebar - end
+        #elseif ($theme == 'social')
+          ## Theme social - start
+          #set ($socialThemeIsSameAuthor = $result.author.equals($socialThemePreviousAuthor))
+          #set ($socialThemePreviousAuthor = $result.author)
+          #set ($socialThemeCollapseClass = '')
+          #if ($socialThemeIsSameAuthor)
+            #set ($socialThemeCollapseClass = 'collapsed')
+          #end
+          &lt;li&gt;
+            &lt;div class="result-container $socialThemeCollapseClass"&gt;
+              #displayAuthorAvatarURL(${result.author})
+              &lt;div class="result-items"&gt;
+                #displayAuthorName(${result.author})
+                &lt;div class="result-item"&gt;
+                  [[image:icon:${ICON_BY_TYPE[$result.type]}||class="result-icon"]]
+                  &lt;span class="result-title"&gt;
+                    &lt;a href="${result.href}"&gt;${result.title_}&lt;/a&gt;
```

Comment:
I cannot see any escaping for titles in the code, so this probably also allows XWiki syntax injection and thus remote code execution via the document's title.